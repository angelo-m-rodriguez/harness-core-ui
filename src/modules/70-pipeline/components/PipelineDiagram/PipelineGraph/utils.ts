import { defaultTo } from 'lodash-es'
import type { PipelineGraphState } from '../types'

const shouldAttachRef = (
  intersectingIndex: number,
  isCurrentChildLast: boolean,
  indexRelativeToParent: number
): boolean => {
  return intersectingIndex === -1 // intersecting index is not set
    ? isCurrentChildLast // current child is not the last child in parallel
    : intersectingIndex === indexRelativeToParent // intersecting index is not equal to indexRelativeToParent (i.e index+1 as parent is 0)
}

const shouldRenderGroupNode = (attachRef: boolean, isCurrentChildLast: boolean): boolean => {
  return attachRef && !isCurrentChildLast // render group node is ref is attached and current node is not the last child
}

const isFirstNodeAGroupNode = (intersectingIndex: number, collapseOnIntersect = false, childLength = 0): boolean =>
  intersectingIndex === 0 && collapseOnIntersect && childLength > 0 // if intersecting index is 0 and collapseOnIntersect is enabled

const showChildNode = (indexRelativeToParent: number, intersectingIndex: number): boolean => {
  return !(indexRelativeToParent > intersectingIndex && intersectingIndex !== -1) // child index is greater than intersectingIndex and intersectingIndex is set to some value
}

const isNodeParallel = (node: PipelineGraphState): boolean => defaultTo(node?.children?.length, 0) > 0

export { shouldAttachRef, shouldRenderGroupNode, isFirstNodeAGroupNode, showChildNode, isNodeParallel }
