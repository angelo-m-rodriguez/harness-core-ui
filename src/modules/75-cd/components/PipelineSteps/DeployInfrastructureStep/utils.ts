/*
 * Copyright 2022 Harness Inc. All rights reserved.
 * Use of this source code is governed by the PolyForm Shield 1.0.0 license
 * that can be found in the licenses directory at the root of this repository, also available at
 * https://polyformproject.org/wp-content/uploads/2020/06/PolyForm-Shield-1.0.0.txt.
 */

import type { FormikProps } from 'formik'
import { isEmpty } from 'lodash-es'

import { getMultiTypeFromValue, MultiTypeInputType, SelectOption } from '@harness/uicore'
import type { EnvironmentResponseDTO, PipelineInfrastructure } from 'services/cd-ng'

import type { StepViewType } from '@pipeline/components/AbstractSteps/Step'

export interface PipelineInfrastructureV2 extends PipelineInfrastructure {
  environmentOrEnvGroupRef?: SelectOption
  environmentGroup?: any
  environmentRef2?: any
  infrastructureRef?: any
}

export interface DeployInfrastructureProps {
  initialValues: PipelineInfrastructureV2
  onUpdate?: (data: PipelineInfrastructureV2) => void
  readonly: boolean
  allowableTypes: MultiTypeInputType[]
  stepViewType?: StepViewType
  inputSetData?: {
    template?: PipelineInfrastructureV2
    path?: string
    readonly?: boolean
  }
}

export interface DeployInfrastructureState {
  isEdit: boolean
  isEnvironment: boolean
  formik?: FormikProps<PipelineInfrastructureV2>
  data?: EnvironmentResponseDTO
}

export function isEditEnvironment(data: PipelineInfrastructureV2): boolean {
  if (
    getMultiTypeFromValue(data.environmentRef) !== MultiTypeInputType.RUNTIME &&
    !isEmpty(data.environmentOrEnvGroupRef)
  ) {
    return true
  }
  return false
}

export function isEditInfrastructure(data: any): boolean {
  if (
    getMultiTypeFromValue(data.infrastructureRef) !== MultiTypeInputType.RUNTIME &&
    !isEmpty(data.infrastructureRef)
  ) {
    return true
  } else if (data.infrastructure && !isEmpty(data.infrastructure.identifier)) {
    return true
  }
  return false
}
