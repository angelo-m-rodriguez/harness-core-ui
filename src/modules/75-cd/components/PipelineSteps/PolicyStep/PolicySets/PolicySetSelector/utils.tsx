/*
 * Copyright 2022 Harness Inc. All rights reserved.
 * Use of this source code is governed by the PolyForm Shield 1.0.0 license
 * that can be found in the licenses directory at the root of this repository, also available at
 * https://polyformproject.org/wp-content/uploads/2020/06/PolyForm-Shield-1.0.0.txt.
 */

import React from 'react'
import { FontVariation, Layout, Text } from '@harness/uicore'
import { useStrings } from 'framework/strings'

export const PolicySetSelectorDialogTitle = (): JSX.Element => {
  const { getString } = useStrings()
  return (
    <Layout.Horizontal
      spacing="xsmall"
      padding={{ top: 'xxlarge', left: 'medium', right: 'large', bottom: 'medium' }}
      flex={{ justifyContent: 'space-between' }}
    >
      <Layout.Vertical>
        <Text font={{ variation: FontVariation.H3 }}>{getString('common.policiesSets.selectPolicySet')}</Text>
      </Layout.Vertical>
    </Layout.Horizontal>
  )
}
