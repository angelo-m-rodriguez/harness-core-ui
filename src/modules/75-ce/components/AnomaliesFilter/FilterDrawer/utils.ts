/*
 * Copyright 2022 Harness Inc. All rights reserved.
 * Use of this source code is governed by the PolyForm Shield 1.0.0 license
 * that can be found in the licenses directory at the root of this repository, also available at
 * https://polyformproject.org/wp-content/uploads/2020/06/PolyForm-Shield-1.0.0.txt.
 */

import type { MultiSelectOption } from '@harness/uicore'
import { omit } from 'lodash-es'
import type { AnomalyFilterProperties } from 'services/ce'
import type { AnomaliesFilterFormType } from './FilterDrawer'

const getMultiSelectOptions = (values: string[]): MultiSelectOption[] =>
  values.map(item => ({ label: item, value: item }))

const getValueFromOption = (values: MultiSelectOption[]): string[] => values.map(item => item.value.toString())

export const getAnomalyFormValuesFromFilterProperties = (
  filterProperties: AnomalyFilterProperties
): AnomaliesFilterFormType => {
  const formValues: AnomaliesFilterFormType = {}

  Object.entries(omit(filterProperties, 'filterType')).forEach(([key, value]) => {
    if (value) {
      if (key === 'minActualAmount' || key === 'minAnomalousSpend') {
        formValues[key] = +value
      } else {
        ;(formValues as any)[key] = getMultiSelectOptions(value as string[])
      }
    }
  })

  return formValues
}

export const getAnomalyFilterPropertiesFromForm = (formData: AnomaliesFilterFormType): AnomalyFilterProperties => {
  const filterProperties: AnomalyFilterProperties = { filterType: 'Anomaly' }

  Object.entries(formData).forEach(([key, value]) => {
    if (key === 'minActualAmount' || key === 'minAnomalousSpend') {
      filterProperties[key] = +value
    } else {
      ;(filterProperties as any)[key] = getValueFromOption(value)
    }
  })

  return filterProperties
}
