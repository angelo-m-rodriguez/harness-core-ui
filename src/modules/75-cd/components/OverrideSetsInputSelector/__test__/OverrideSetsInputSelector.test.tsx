/*
 * Copyright 2022 Harness Inc. All rights reserved.
 * Use of this source code is governed by the PolyForm Shield 1.0.0 license
 * that can be found in the licenses directory at the root of this repository, also available at
 * https://polyformproject.org/wp-content/uploads/2020/06/PolyForm-Shield-1.0.0.txt.
 */

import React from 'react'
import { render, fireEvent, act } from '@testing-library/react'
import type { DeploymentStageElementConfig, StageElementWrapper } from '@pipeline/utils/pipelineTypes'
import { findPopoverContainer } from '@common/utils/testUtils'
import {
  PipelineContext,
  PipelineContextInterface
} from '@pipeline/components/PipelineStudio/PipelineContext/PipelineContext'
import { OverrideSetsInputSelector } from '../OverrideSetsInputSelector'

jest.mock('framework/strings', () => ({
  useStrings: () => ({
    getString: (key: string) => key
  })
}))
const InputSetSelectorProps = {
  value: [
    {
      label: 'demo',
      value: 'demoValue'
    },
    {
      label: 'demo1',
      value: 'demoValue1'
    }
  ],
  width: 350
}

const stageMock: StageElementWrapper<DeploymentStageElementConfig> = {
  stage: {
    spec: {
      execution: {
        rollbackSteps: undefined,
        steps: []
      },
      infrastructure: {
        allowSimultaneousDeployments: false,
        environment: undefined,
        environmentRef: 'environmentRef',
        infrastructureDefinition: undefined,
        infrastructureKey: 'infrastructureKey',
        useFromStage: undefined
      },
      serviceConfig: {
        service: undefined,
        serviceDefinition: {
          type: 'Kubernetes',
          spec: {
            artifacts: {
              sidecars: [
                {
                  sidecar: {
                    identifier: 'id11',
                    type: 'Gcr',
                    spec: {
                      connectorRef: 'gcp_connector',
                      registryHostname: 'gcr.io',
                      imagePath: 'library/nginx',
                      tag: '<+input>'
                    }
                  }
                },
                {
                  sidecar: {
                    identifier: 'id11',
                    type: 'DockerRegistry',
                    spec: {
                      connectorRef: 'harnessImage',
                      imagePath: 'library/nginx',
                      tag: '<+input>'
                    }
                  }
                }
              ],
              primary: {
                type: 'DockerRegistry',
                spec: { connectorRef: '<+input>', imagePath: '<+input>' }
              }
            },
            manifests: [],
            artifactOverrideSets: [
              {
                overrideSet: {
                  identifier: 'overrideSetIdentifier',
                  artifacts: {
                    sidecars: [],
                    primary: {
                      type: 'DockerRegistry',
                      spec: { connectorRef: '<+input>', imagePath: '<+input>' }
                    }
                  }
                }
              }
            ],
            manifestOverrideSets: [
              {
                overrideSet: {
                  identifier: 'overrideSetIdentifier'
                }
              }
            ],
            variableOverrideSets: [
              {
                overrideSet: {
                  identifier: 'overrideSetIdentifier'
                }
              }
            ]
          }
        },
        serviceRef: 'serviceRef',
        stageOverrides: {
          artifacts: undefined,
          manifests: undefined,
          variables: undefined,
          useArtifactOverrideSets: ['artifact1', 'artifact2'],
          useManifestOverrideSets: ['manifest1'],
          useVariableOverrideSets: ['variable1', 'variable2']
        },
        useFromStage: {
          stage: 'deploy'
        }
      }
    },
    identifier: 'identifier',
    name: 'name'
  }
}

const getContextValue = (): PipelineContextInterface => {
  return {
    ...stageMock,
    state: {
      selectionState: { selectedStageId: 'selectedID' }
    },
    getStageFromPipeline: jest.fn(() => {
      return { stage: stageMock, parent: undefined }
    })
  } as any
}

describe('OverrideSetsInputSelector ', () => {
  test('should render Artifact context', async () => {
    const { container, getByText, getByPlaceholderText } = render(
      <PipelineContext.Provider value={getContextValue()}>
        <OverrideSetsInputSelector context={'ARTIFACT'} onChange={jest.fn()} value={InputSetSelectorProps.value} />
      </PipelineContext.Provider>
    )
    fireEvent.click(container.querySelector('[data-icon="caret-down"]') as HTMLElement)
    const popover = findPopoverContainer()
    expect(popover).toMatchSnapshot()

    //checkbox Handler
    await act(async () => {
      const checkBoxOverride = fireEvent.click(getByText('overrideSetIdentifier'))
      fireEvent.click(document.querySelector('input[type=checkbox]') as HTMLElement)
      expect(checkBoxOverride).toBeTruthy()
    })

    // checkBox change
    fireEvent.click(document.querySelector('.item') as HTMLElement)
    const checkBoxDemo = fireEvent.click(getByText('demo'))
    expect(checkBoxDemo).toBeTruthy()

    //searchBox Change
    const searchBox = getByPlaceholderText('overrideSet.searchInputSet')
    expect(searchBox).not.toBe(null)
    act(() => {
      fireEvent.change(getByPlaceholderText('overrideSet.searchInputSet'), { target: { value: 'randomstring223' } })
    })
    expect(container).toMatchSnapshot()
  })
  test('should render Manifest context', () => {
    const { container } = render(
      <PipelineContext.Provider value={getContextValue()}>
        <OverrideSetsInputSelector context={'MANIFEST'} onChange={jest.fn()} value={InputSetSelectorProps.value} />
      </PipelineContext.Provider>
    )
    expect(container).toMatchSnapshot()
  })
  test('should render Variables context', () => {
    const { container } = render(
      <PipelineContext.Provider value={getContextValue()}>
        <OverrideSetsInputSelector context={'VARIABLES'} onChange={jest.fn()} value={InputSetSelectorProps.value} />
      </PipelineContext.Provider>
    )
    expect(container).toMatchSnapshot()
  })
  test('should render Artifact and value not array', async () => {
    const { container } = render(
      <OverrideSetsInputSelector
        context={'ARTIFACT'}
        onChange={jest.fn()}
        value={{ label: 'singleValue', value: 'demoSingle' }}
      />
    )
    fireEvent.click(container.querySelector('[data-icon="cross"]') as HTMLElement)
    expect(container).toMatchSnapshot()
  })
  test('should render Artifact no value', () => {
    const { container } = render(<OverrideSetsInputSelector context={'ARTIFACT'} onChange={jest.fn()} />)
    expect(container).toMatchSnapshot()
  })
})
