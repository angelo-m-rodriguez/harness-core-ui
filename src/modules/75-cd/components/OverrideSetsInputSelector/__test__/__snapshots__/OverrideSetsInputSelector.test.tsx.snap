// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`OverrideSetsInputSelector  should render Artifact and value not array 1`] = `
<div>
  <span
    class="bp3-popover-wrapper"
  >
    <span
      class="bp3-popover-target"
    >
      <button
        class="bp3-button bp3-minimal Button--button StyledProps--font StyledProps--main container Button--with-current-color Button--iconOnly Button--without-shadow Button--withRightIcon"
        style="width: 350px;"
        type="button"
      >
        <span
          class="bp3-button-text"
        >
          <div
            class="Layout--horizontal StyledProps--main StyledProps--flex StyledProps--flex-distribution-space-between"
          >
            <span>
              singleValue
            </span>
            <button
              class="clearButton"
            >
              <span
                class="bp3-icon bp3-icon-cross StyledProps--main"
                icon="cross"
              >
                <svg
                  data-icon="cross"
                  height="16"
                  viewBox="0 0 16 16"
                  width="16"
                >
                  <desc>
                    cross
                  </desc>
                  <path
                    d="M9.41 8l3.29-3.29c.19-.18.3-.43.3-.71a1.003 1.003 0 00-1.71-.71L8 6.59l-3.29-3.3a1.003 1.003 0 00-1.42 1.42L6.59 8 3.3 11.29c-.19.18-.3.43-.3.71a1.003 1.003 0 001.71.71L8 9.41l3.29 3.29c.18.19.43.3.71.3a1.003 1.003 0 00.71-1.71L9.41 8z"
                    fill-rule="evenodd"
                  />
                </svg>
              </span>
            </button>
          </div>
        </span>
        <span
          class="bp3-icon bp3-icon-caret-down StyledProps--main"
          icon="caret-down"
        >
          <svg
            data-icon="caret-down"
            height="16"
            viewBox="0 0 16 16"
            width="16"
          >
            <desc>
              caret-down
            </desc>
            <path
              d="M12 6.5c0-.28-.22-.5-.5-.5h-7a.495.495 0 00-.37.83l3.5 4c.09.1.22.17.37.17s.28-.07.37-.17l3.5-4c.08-.09.13-.2.13-.33z"
              fill-rule="evenodd"
            />
          </svg>
        </span>
      </button>
    </span>
  </span>
</div>
`;

exports[`OverrideSetsInputSelector  should render Artifact context 1`] = `
<div
  class="bp3-popover-content"
>
  <div
    class="Layout--vertical Layout--layout-spacing-small StyledProps--main popoverContainer StyledProps--padding-medium"
  >
    <div
      class="Layout--horizontal Layout--layout-spacing-medium StyledProps--main"
    >
      <div
        class="TextInput--main"
      >
        <div
          class="bp3-input-group search"
        >
          <span
            class="bp3-icon bp3-icon-search StyledProps--main"
            icon="search"
          >
            <svg
              data-icon="search"
              height="12"
              viewBox="0 0 16 16"
              width="12"
            >
              <desc>
                search
              </desc>
              <path
                d="M15.55 13.43l-2.67-2.68a6.94 6.94 0 001.11-3.76c0-3.87-3.13-7-7-7s-7 3.13-7 7 3.13 7 7 7c1.39 0 2.68-.42 3.76-1.11l2.68 2.67a1.498 1.498 0 102.12-2.12zm-8.56-1.44c-2.76 0-5-2.24-5-5s2.24-5 5-5 5 2.24 5 5-2.24 5-5 5z"
                fill-rule="evenodd"
              />
            </svg>
          </span>
          <input
            class="bp3-input"
            placeholder="overrideSet.searchInputSet"
            type="text"
            value=""
          />
        </div>
      </div>
      <label
        class="bp3-control bp3-checkbox StyledProps--font Checkbox--checkbox checkbox StyledProps--main"
      >
        <input
          checked=""
          type="checkbox"
        />
        <span
          class="bp3-control-indicator"
        />
        overrideSet.selectMultipleInputSets
      </label>
    </div>
    <div
      class="Layout--horizontal Layout--layout-spacing-small StyledProps--main StyledProps--background-grey200 StyledProps--flex StyledProps--flex-align-center-center StyledProps--padding-small"
    >
      <span
        class="bp3-icon bp3-icon-info-sign StyledProps--main StyledProps--color StyledProps--color-primary4"
        icon="info-sign"
      >
        <svg
          data-icon="info-sign"
          height="16"
          viewBox="0 0 16 16"
          width="16"
        >
          <desc>
            info-sign
          </desc>
          <path
            d="M8 0C3.58 0 0 3.58 0 8s3.58 8 8 8 8-3.58 8-8-3.58-8-8-8zM7 3h2v2H7V3zm3 10H6v-1h1V7H6V6h3v6h1v1z"
            fill-rule="evenodd"
          />
        </svg>
      </span>
      <p
        class="StyledProps--font StyledProps--main StyledProps--font-size-small"
      >
        overrideSet.helpTextForMultiSelect
      </p>
    </div>
    <div
      class="Layout--vertical StyledProps--main"
    >
      <div
        class="Layout--horizontal StyledProps--main StyledProps--padding-small StyledProps--flex StyledProps--flex-distribution-space-between"
      >
        <p
          class="StyledProps--font StyledProps--main StyledProps--color StyledProps--color-black"
        >
          overrideSet.inputSet
        </p>
        <p
          class="StyledProps--font StyledProps--main StyledProps--color StyledProps--color-black"
        >
          overrideSet.order
        </p>
      </div>
      <ul
        class="bp3-menu list multiple"
      >
        <li
          class="item"
          draggable="true"
        >
          <div
            class="Layout--horizontal StyledProps--main StyledProps--flex StyledProps--flex-distribution-space-between"
          >
            <label
              class="bp3-control bp3-checkbox StyledProps--font Checkbox--checkbox checkbox StyledProps--main"
            >
              <input
                checked=""
                type="checkbox"
              />
              <span
                class="bp3-control-indicator"
              />
              <p
                class="StyledProps--font StyledProps--main StyledProps--font-size-small"
                style="display: flex; align-items: center;"
              >
                <span
                  class="bp3-icon StyledProps--main Text--icon StyledProps--padding-right-xsmall"
                  data-icon="yaml-builder-input-sets"
                >
                  <svg
                    height="16"
                    viewBox="0 0 18 16"
                    width="16"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <g
                      fill="#22272D"
                    >
                      <path
                        d="M12.946 5.347a.49.49 0 00-.492-.486H4.581a.49.49 0 00-.492.486c0 .269.22.486.492.486h7.873a.49.49 0 00.492-.486zM4.581 8.264c-.272 0-.492.218-.492.486s.22.486.492.486H9.01c.271 0 .492-.218.492-.486s-.22-.486-.492-.486H4.58zm7.92 5.833H3.644a.55.55 0 01-.538-.56V2.018a.55.55 0 01.538-.56h9.748a.55.55 0 01.54.56v5.325c0 .269.22.486.491.486a.49.49 0 00.492-.486V2.02c0-.844-.679-1.533-1.523-1.533H3.643c-.844 0-1.523.69-1.523 1.533v11.517c0 .844.68 1.533 1.523 1.533h8.859a.49.49 0 00.492-.486.49.49 0 00-.492-.486z"
                      />
                      <path
                        d="M15.676 10.418h-1.005V9.413a.524.524 0 10-1.048 0v1.004H12.28a.524.524 0 000 1.049h1.343v1.343a.524.524 0 101.048 0v-1.343h1.005a.524.524 0 000-1.048z"
                      />
                    </g>
                  </svg>
                </span>
                demo
              </p>
            </label>
            <span
              class="order"
            >
              <p
                class="StyledProps--font StyledProps--main orderText"
              >
                1
              </p>
              <span
                class="bp3-icon StyledProps--main"
                data-icon="main-reorder"
              >
                <svg
                  height="12"
                  viewBox="0 0 12 13"
                  width="12"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M6 0c-.1 0-.2 0-.3.1L4.1 1.6c-.2.2-.1.6.3.6h3.2c.4 0 .5-.4.3-.6L6.3.1C6.2 0 6.1 0 6 0zM.7 3.4c-.2 0-.5.1-.6.3s-.1.5 0 .6c.1.2.3.3.6.3h10.6c.2 0 .5-.1.6-.3s.1-.4 0-.6-.3-.3-.6-.3zm0 2.5c-.3 0-.5.1-.6.3s-.1.4 0 .6.3.3.6.3h10.6c.2 0 .5-.1.6-.3s.1-.4 0-.6-.3-.3-.6-.3zm0 2.4c-.2 0-.5.1-.6.3s-.1.5 0 .7.3.3.6.3h10.6c.2 0 .5-.1.6-.3s.1-.4 0-.6-.3-.3-.6-.3H.7zm3.7 2.5c-.4 0-.5.4-.3.6l1.5 1.4c.2.2.5.2.7 0l1.5-1.4c.2-.2.1-.6-.3-.6z"
                    fill="#4d5969"
                  />
                </svg>
              </span>
            </span>
          </div>
        </li>
        <li
          class="item"
          draggable="true"
        >
          <div
            class="Layout--horizontal StyledProps--main StyledProps--flex StyledProps--flex-distribution-space-between"
          >
            <label
              class="bp3-control bp3-checkbox StyledProps--font Checkbox--checkbox checkbox StyledProps--main"
            >
              <input
                checked=""
                type="checkbox"
              />
              <span
                class="bp3-control-indicator"
              />
              <p
                class="StyledProps--font StyledProps--main StyledProps--font-size-small"
                style="display: flex; align-items: center;"
              >
                <span
                  class="bp3-icon StyledProps--main Text--icon StyledProps--padding-right-xsmall"
                  data-icon="yaml-builder-input-sets"
                >
                  <svg
                    height="16"
                    viewBox="0 0 18 16"
                    width="16"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <g
                      fill="#22272D"
                    >
                      <path
                        d="M12.946 5.347a.49.49 0 00-.492-.486H4.581a.49.49 0 00-.492.486c0 .269.22.486.492.486h7.873a.49.49 0 00.492-.486zM4.581 8.264c-.272 0-.492.218-.492.486s.22.486.492.486H9.01c.271 0 .492-.218.492-.486s-.22-.486-.492-.486H4.58zm7.92 5.833H3.644a.55.55 0 01-.538-.56V2.018a.55.55 0 01.538-.56h9.748a.55.55 0 01.54.56v5.325c0 .269.22.486.491.486a.49.49 0 00.492-.486V2.02c0-.844-.679-1.533-1.523-1.533H3.643c-.844 0-1.523.69-1.523 1.533v11.517c0 .844.68 1.533 1.523 1.533h8.859a.49.49 0 00.492-.486.49.49 0 00-.492-.486z"
                      />
                      <path
                        d="M15.676 10.418h-1.005V9.413a.524.524 0 10-1.048 0v1.004H12.28a.524.524 0 000 1.049h1.343v1.343a.524.524 0 101.048 0v-1.343h1.005a.524.524 0 000-1.048z"
                      />
                    </g>
                  </svg>
                </span>
                demo1
              </p>
            </label>
            <span
              class="order"
            >
              <p
                class="StyledProps--font StyledProps--main orderText"
              >
                2
              </p>
              <span
                class="bp3-icon StyledProps--main"
                data-icon="main-reorder"
              >
                <svg
                  height="12"
                  viewBox="0 0 12 13"
                  width="12"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M6 0c-.1 0-.2 0-.3.1L4.1 1.6c-.2.2-.1.6.3.6h3.2c.4 0 .5-.4.3-.6L6.3.1C6.2 0 6.1 0 6 0zM.7 3.4c-.2 0-.5.1-.6.3s-.1.5 0 .6c.1.2.3.3.6.3h10.6c.2 0 .5-.1.6-.3s.1-.4 0-.6-.3-.3-.6-.3zm0 2.5c-.3 0-.5.1-.6.3s-.1.4 0 .6.3.3.6.3h10.6c.2 0 .5-.1.6-.3s.1-.4 0-.6-.3-.3-.6-.3zm0 2.4c-.2 0-.5.1-.6.3s-.1.5 0 .7.3.3.6.3h10.6c.2 0 .5-.1.6-.3s.1-.4 0-.6-.3-.3-.6-.3H.7zm3.7 2.5c-.4 0-.5.4-.3.6l1.5 1.4c.2.2.5.2.7 0l1.5-1.4c.2-.2.1-.6-.3-.6z"
                    fill="#4d5969"
                  />
                </svg>
              </span>
            </span>
          </div>
        </li>
        <li
          class="item"
        >
          <div
            class="Layout--horizontal StyledProps--main StyledProps--flex StyledProps--flex-distribution-space-between"
          >
            <label
              class="bp3-control bp3-checkbox StyledProps--font Checkbox--checkbox checkbox StyledProps--main"
            >
              <input
                type="checkbox"
              />
              <span
                class="bp3-control-indicator"
              />
              <p
                class="StyledProps--font StyledProps--main StyledProps--font-size-small"
                style="display: flex; align-items: center;"
              >
                <span
                  class="bp3-icon StyledProps--main Text--icon StyledProps--padding-right-xsmall"
                  data-icon="yaml-builder-input-sets"
                >
                  <svg
                    height="16"
                    viewBox="0 0 18 16"
                    width="16"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <g
                      fill="#22272D"
                    >
                      <path
                        d="M12.946 5.347a.49.49 0 00-.492-.486H4.581a.49.49 0 00-.492.486c0 .269.22.486.492.486h7.873a.49.49 0 00.492-.486zM4.581 8.264c-.272 0-.492.218-.492.486s.22.486.492.486H9.01c.271 0 .492-.218.492-.486s-.22-.486-.492-.486H4.58zm7.92 5.833H3.644a.55.55 0 01-.538-.56V2.018a.55.55 0 01.538-.56h9.748a.55.55 0 01.54.56v5.325c0 .269.22.486.491.486a.49.49 0 00.492-.486V2.02c0-.844-.679-1.533-1.523-1.533H3.643c-.844 0-1.523.69-1.523 1.533v11.517c0 .844.68 1.533 1.523 1.533h8.859a.49.49 0 00.492-.486.49.49 0 00-.492-.486z"
                      />
                      <path
                        d="M15.676 10.418h-1.005V9.413a.524.524 0 10-1.048 0v1.004H12.28a.524.524 0 000 1.049h1.343v1.343a.524.524 0 101.048 0v-1.343h1.005a.524.524 0 000-1.048z"
                      />
                    </g>
                  </svg>
                </span>
                overrideSetIdentifier
              </p>
            </label>
          </div>
        </li>
      </ul>
    </div>
  </div>
</div>
`;

exports[`OverrideSetsInputSelector  should render Artifact context 2`] = `
<div>
  <span
    class="bp3-popover-wrapper"
  >
    <span
      class="bp3-popover-target bp3-popover-open"
    >
      <button
        class="bp3-button bp3-minimal Button--button StyledProps--font StyledProps--main container bp3-active Button--with-current-color Button--iconOnly Button--without-shadow Button--withRightIcon"
        style="width: 350px;"
        type="button"
      >
        <span
          class="bp3-button-text"
        >
          <span
            class="Tag--main"
          >
            <span
              class="bp3-tag bp3-minimal tag"
            >
              <span
                class="bp3-text-overflow-ellipsis bp3-fill"
              >
                demo
              </span>
            </span>
          </span>
          <span
            class="Tag--main"
          >
            <span
              class="bp3-tag bp3-minimal tag"
            >
              <span
                class="bp3-text-overflow-ellipsis bp3-fill"
              >
                demo1
              </span>
            </span>
          </span>
        </span>
        <span
          class="bp3-icon bp3-icon-caret-down StyledProps--main"
          icon="caret-down"
        >
          <svg
            data-icon="caret-down"
            height="16"
            viewBox="0 0 16 16"
            width="16"
          >
            <desc>
              caret-down
            </desc>
            <path
              d="M12 6.5c0-.28-.22-.5-.5-.5h-7a.495.495 0 00-.37.83l3.5 4c.09.1.22.17.37.17s.28-.07.37-.17l3.5-4c.08-.09.13-.2.13-.33z"
              fill-rule="evenodd"
            />
          </svg>
        </span>
      </button>
    </span>
  </span>
</div>
`;

exports[`OverrideSetsInputSelector  should render Artifact no value 1`] = `
<div>
  <span
    class="bp3-popover-wrapper"
  >
    <span
      class="bp3-popover-target"
    >
      <button
        class="bp3-button bp3-minimal Button--button StyledProps--font StyledProps--main container Button--with-current-color Button--iconOnly Button--without-shadow Button--withRightIcon"
        style="width: 350px;"
        type="button"
      >
        <span
          class="bp3-button-text"
        >
          <span
            class="placeholder"
          >
            overrideSet.placeholder
          </span>
        </span>
        <span
          class="bp3-icon bp3-icon-caret-down StyledProps--main"
          icon="caret-down"
        >
          <svg
            data-icon="caret-down"
            height="16"
            viewBox="0 0 16 16"
            width="16"
          >
            <desc>
              caret-down
            </desc>
            <path
              d="M12 6.5c0-.28-.22-.5-.5-.5h-7a.495.495 0 00-.37.83l3.5 4c.09.1.22.17.37.17s.28-.07.37-.17l3.5-4c.08-.09.13-.2.13-.33z"
              fill-rule="evenodd"
            />
          </svg>
        </span>
      </button>
    </span>
  </span>
</div>
`;

exports[`OverrideSetsInputSelector  should render Manifest context 1`] = `
<div>
  <span
    class="bp3-popover-wrapper"
  >
    <span
      class="bp3-popover-target"
    >
      <button
        class="bp3-button bp3-minimal Button--button StyledProps--font StyledProps--main container Button--with-current-color Button--iconOnly Button--without-shadow Button--withRightIcon"
        style="width: 350px;"
        type="button"
      >
        <span
          class="bp3-button-text"
        >
          <span
            class="Tag--main"
          >
            <span
              class="bp3-tag bp3-minimal tag"
            >
              <span
                class="bp3-text-overflow-ellipsis bp3-fill"
              >
                demo
              </span>
            </span>
          </span>
          <span
            class="Tag--main"
          >
            <span
              class="bp3-tag bp3-minimal tag"
            >
              <span
                class="bp3-text-overflow-ellipsis bp3-fill"
              >
                demo1
              </span>
            </span>
          </span>
        </span>
        <span
          class="bp3-icon bp3-icon-caret-down StyledProps--main"
          icon="caret-down"
        >
          <svg
            data-icon="caret-down"
            height="16"
            viewBox="0 0 16 16"
            width="16"
          >
            <desc>
              caret-down
            </desc>
            <path
              d="M12 6.5c0-.28-.22-.5-.5-.5h-7a.495.495 0 00-.37.83l3.5 4c.09.1.22.17.37.17s.28-.07.37-.17l3.5-4c.08-.09.13-.2.13-.33z"
              fill-rule="evenodd"
            />
          </svg>
        </span>
      </button>
    </span>
  </span>
</div>
`;

exports[`OverrideSetsInputSelector  should render Variables context 1`] = `
<div>
  <span
    class="bp3-popover-wrapper"
  >
    <span
      class="bp3-popover-target"
    >
      <button
        class="bp3-button bp3-minimal Button--button StyledProps--font StyledProps--main container Button--with-current-color Button--iconOnly Button--without-shadow Button--withRightIcon"
        style="width: 350px;"
        type="button"
      >
        <span
          class="bp3-button-text"
        >
          <span
            class="Tag--main"
          >
            <span
              class="bp3-tag bp3-minimal tag"
            >
              <span
                class="bp3-text-overflow-ellipsis bp3-fill"
              >
                demo
              </span>
            </span>
          </span>
          <span
            class="Tag--main"
          >
            <span
              class="bp3-tag bp3-minimal tag"
            >
              <span
                class="bp3-text-overflow-ellipsis bp3-fill"
              >
                demo1
              </span>
            </span>
          </span>
        </span>
        <span
          class="bp3-icon bp3-icon-caret-down StyledProps--main"
          icon="caret-down"
        >
          <svg
            data-icon="caret-down"
            height="16"
            viewBox="0 0 16 16"
            width="16"
          >
            <desc>
              caret-down
            </desc>
            <path
              d="M12 6.5c0-.28-.22-.5-.5-.5h-7a.495.495 0 00-.37.83l3.5 4c.09.1.22.17.37.17s.28-.07.37-.17l3.5-4c.08-.09.13-.2.13-.33z"
              fill-rule="evenodd"
            />
          </svg>
        </span>
      </button>
    </span>
  </span>
</div>
`;
